project-koku/koku
# COST-267: Fix exclusion on org endpoint to not exclude dates equal to end date.

Issue: https://issues.redhat.com/browse/COST-267

We noticed that when logged in under the qa account in ci that the data under the organizations [endpoint](https://ci.cloud.redhat.com/api/cost-management/v1/organizations/aws/) was coming back as an `[]`. After some debugging, we discovered that the issue was on this line here: 

https://github.com/project-koku/koku/blob/master/koku/api/organizations/queries.py#L265

```
org_unit_query = org_unit_query.exclude(created_timestamp__gte=self.end_datetime)
```

I am excluding org units based on if the created timestamp is greater or equal to the end_datetime, since the end datetime is defaulted to today: `self.end_datetime = self.dh.today`, I was excluding the `6-17-2020` data which is the only data (which is the only data we have in the ci for the qa account). Therefore, this issue is to not exclude org units and accounts that are created on the same day as the end date. 

# To Test:

So, I changed the scripting underneath `make load-aws-org-unit-tree` for this pr so that `OU_005` is created on whatever today is:
```
postgres=# SELECT * FROM acct10001.reporting_awsorganizationalunit;
 id | org_unit_name | org_unit_id |    org_unit_path    | level | created_timestamp | deleted_timestamp | account_alias_id
----+---------------+-------------+---------------------+-------+-------------------+-------------------+------------------
  1 | Root          | R_001       | R_001               |     0 | 2020-03-19        |                   |
  2 | Root          | R_001       | R_001               |     0 | 2020-03-19        |                   |                1
  3 | Dept OU_001   | OU_001      | R_001&OU_001        |     1 | 2020-03-19        |                   |
  4 | Dept OU_001   | OU_001      | R_001&OU_001        |     1 | 2020-03-19        |                   |                2
  5 | Dept OU_001   | OU_001      | R_001&OU_001        |     1 | 2020-03-19        |                   |                3
  6 | Dept OU_002   | OU_002      | R_001&OU_002        |     1 | 2020-03-19        |                   |
  7 | Dept OU_002   | OU_002      | R_001&OU_002        |     1 | 2020-03-19        |                   |                4
  8 | Dept OU_003   | OU_003      | R_001&OU_002&OU_003 |     2 | 2020-03-19        |                   |
 11 | Dept OU_005   | OU_005      | R_001&OU_002&OU_005 |     2 | 2020-03-19        |                   |
  9 | Dept OU_004   | OU_004      | R_001&OU_004        |     1 | 2020-03-19        | 2020-05-17        |
 10 | Dept OU_004   | OU_004      | R_001&OU_004        |     1 | 2020-03-19        | 2020-05-17        |                5
 12 | Dept OU_003   | OU_003      | R_001&OU_002&OU_003 |     2 | 2020-06-17        |                   |                4
 13 | Dept OU_005   | OU_005      | R_001&OU_001&OU_005 |     2 | 2020-06-17        |                   |
```

Therefore, to test all we need to do is pull this branch, run `make load-aws-org-unit-tree` and make sure that the OU_005 still shows up in the: http://127.0.0.1:8000/api/cost-management/v1/organizations/aws/ endpoint.
